livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterpreter.java

```diff
@@ -331,7 +331,13 @@ private InterpreterResult appendSessionExpire(InterpreterResult result, boolean
   private InterpreterResult getResultFromStatementInfo(StatementInfo stmtInfo,
                                                        boolean displayAppInfo) {
     if (stmtInfo.output != null && stmtInfo.output.isError()) {
-      return new InterpreterResult(InterpreterResult.Code.ERROR, stmtInfo.output.evalue);
+      InterpreterResult result = new InterpreterResult(InterpreterResult.Code.ERROR);
+      result.add(stmtInfo.output.evalue);
+      if (stmtInfo.output.traceback != null) {
+        for (int i = 0; i < stmtInfo.output.traceback.length; i++)
+          result.add(stmtInfo.output.traceback[i]);
```

Comment:
I see. The extra line is caused by the \n within each line of the traceback. Join all of the lines to one message do seems to be a better  idea.